kg_microbe/transform_utils/madin_etal/madin_etal.py

```diff
@@ -244,49 +257,94 @@ def run(self, data_file: Union[Optional[Path], Optional[str]] = None, show_statu
+                                        metpo_mapping["label"],
+                                    ]
                                 )
                                 tax_pathway_edge.append(
                                     [
                                         tax_id,
-                                        NCBI_TO_PATHWAY_EDGE,
-                                        row[1].object_id,
+                                        predicate,
+                                        metpo_mapping["curie"],
                                         BIOLOGICAL_PROCESS,
                                     ]
                                 )
+                            else:
+                                pathways_not_in_metpo.append(pathway)
+
+                        # For pathways not found in METPO, fall back to NER results
+                        if pathways_not_in_metpo:
+                            go_condition = go_result[TRAITS_DATASET_LABEL_COLUMN].isin(
+                                pathways_not_in_metpo
+                            )
+                            go_result_for_tax_id = go_result.loc[go_condition]
+                            if go_result_for_tax_id.empty:
+                                # Use fallback naming if no NER results
+                                for item in pathways_not_in_metpo:
+                                    pathway_nodes.append(
+                                        [
+                                            PATHWAY_PREFIX + item.strip(),
+                                            PATHWAY_CATEGORY,
+                                            item.strip(),
+                                        ]
+                                    )
+                                    tax_pathway_edge.append(
+                                        [
+                                            tax_id,
+                                            NCBI_TO_PATHWAY_EDGE,
+                                            PATHWAY_PREFIX + item.strip().lower(),
+                                            BIOLOGICAL_PROCESS,
+                                        ]
+                                    )
+                            else:
+                                exact_condition_go = (
+                                    go_result_for_tax_id[OBJECT_LABEL_COLUMN]
+                                    == go_result_for_tax_id[SUBJECT_LABEL_COLUMN]
+                                )
+                                exact_match_go_df = go_result_for_tax_id[exact_condition_go]
+                                if not exact_match_go_df.empty:
+                                    go_result_for_tax_id = exact_match_go_df
+                                for row in go_result_for_tax_id.iterrows():
+                                    pathway_nodes.append(
+                                        [row[1].object_id, PATHWAY_CATEGORY, row[1].object_label]
```

Comment:
@sujaypatil96 here it looks like still using the original PATHWAY_CATEGORY? should this also be coming from the METPO sheet?
It may be that biolink categories are missing in the sheet, I could help with that.

Reply:
This is code in the else handler – if there is no mapping to the _pathways_ column in the METPO sheet.